src/raftstore/store/snap.rs

```diff
@@ -1207,6 +1196,38 @@ impl SnapManager {
         key: &SnapKey,
         snap: &DbSnapshot,
     ) -> RaftStoreResult<Box<Snapshot>> {
+        let mut old_snaps = None;
+        while self.get_total_snap_size() > self.max_total_snap_size() {
+            if old_snaps.is_none() {
+                let snaps = match self.list_idle_snap() {
+                    Ok(snaps) => snaps,
+                    Err(_) => break,
+                };
+                let mut key_and_snaps = Vec::with_capacity(snaps.len());
+                for (key, is_sending) in snaps {
+                    if !is_sending {
+                        continue;
+                    }
+                    let snap = match self.get_snapshot_for_sending(&key) {
+                        Ok(snap) => snap,
+                        Err(_) => continue,
+                    };
+                    if !snap.meta().and_then(|m| m.modified()).is_ok() {
```

Comment:
Store modified time is not conflict with storing the snapshot. Store modified time can reduce IO access during comparison.